plugins/quota/validate.go

```diff
@@ -3,38 +3,71 @@ package quota
 import (
 	"fmt"
 
+	daemon "github.com/gluster/glusterd2/glusterd2/daemon"
 	"github.com/gluster/glusterd2/glusterd2/volume"
 	"github.com/gluster/glusterd2/glusterd2/xlator"
+	"github.com/gluster/glusterd2/pkg/errors"
 )
 
 const (
-	quotaEnabledKey = "features.quota"
+	quotaEnabledKey = "quota.enable"
 )
 
-func validateOptions(v *volume.Volinfo, key string, value string) error {
+func validateOptions(v *volume.Volinfo, key, value string) error {
 
-	if isQuotaEnabled(v) == false {
-		err := fmt.Errorf("Quota not enabled to set this value: '%s'", key)
-		return err
+	// Check if volume is started
+	if v.State != volume.VolStarted {
+		return errors.ErrVolNotStarted
 	}
 
 	switch key {
-	case "deem-statfs":
+	case "enable":
+		switch value {
+		case "on":
+			running := isQuotadRunning()
+			if running {
+				// As quotad for various process share a
+				// single pid, it is necessary to check the
+				// options too.
+				if isQuotaEnabled(v) {
+					return errors.ErrProcessAlreadyRunning
+				}
+
+			}
+		case "off":
+			running := isQuotadRunning()
+			if !running {
+				return errors.ErrProcessNotRunning
+			}
+			// As quotad for various process share a
+			// single pid, it is necessary to check the
+			// options too.
+			if !isQuotaEnabled(v) {
```

Comment:
question:
I think checking options should have the highest priority
first, we should check ```isQuotaEnabled``` and then we should check ``` isQuotadRunning```

Reply:
I think this is fine, as quotad is shared between a number of volumes in the same node. and if the quotad is brought down because of some other volume or action and if we want to start it, this is better. we check if its not running then we start it. we done worry about the option which is "enabled". If you have a better case, do let me know. we will discuss and decide which one to have at the top.